llama-index-core/llama_index/core/node_parser/text/semantic_double_merging_splitter.py

```diff
@@ -248,7 +253,8 @@ def _create_initial_chunks(self, sentences: List[str]) -> List[str]:
                     self.language_config.nlp(self._clean_text_advanced(sentence))
                 )
                 > self.appending_threshold
-                and not len(chunk) > self.max_chunk_size
+                and len(last_sentences) + len(sentence) + 1 <= self.max_chunk_size
```

Comment:
I did not realize this the first time, but `len()` is counting characters, not tokens

We should use a tokenizer to count

```
from llama_index.core import get_tokenizer
tokenizer = get_tokenizer()

chunk_size = len(tokenizer(chunk))
```

Reply:
well, ok I guess counting characters is intended (reading the attribute description), but it is a little misleading. Fine for now